library/gcse/src/gcse_api.e

```diff
@@ -0,0 +1,243 @@
+			-- orTerms={orTerms?}&relatedSite={relatedSite?}&dateRestrict={dateRestrict?}&lowRange={lowRange?}&highRange={highRange?}&
+			-- searchType={searchType}&fileType={fileType?}&rights={rights?}&imgSize={imgSize?}&imgType={imgType?}&
+			-- imgColorType={imgColorType?}&imgDominantColor={imgDominantColor?}&alt=json"
+
+		do
+			create Result.make_from_string (base_uri)
+			Result.append ("?key=")
+			Result.append (query_paremeter.secret)
+			Result.append ("&cx=")
+			Result.append (query_paremeter.cx)
+			Result.append ("&q=")
+			Result.append (query_paremeter.query)
+				-- num
+			if attached query_paremeter.num as l_num then
+				Result.append ("&num=")
+				Result.append (l_num)
+			end
+			if attached query_paremeter.start as l_start then
+				Result.append ("&start=")
+				Result.append (l_start)
+			end
+		end
+
+	put_error (a_code: READABLE_STRING_GENERAL)
+			-- put error with code `a_code'.
+		local
+			l_errors: like errors
+			utf: UTF_CONVERTER
+		do
+			l_errors := errors
+			if l_errors = Void then
+				create {ARRAYED_LIST [STRING]} l_errors.make (1)
+				errors := l_errors
+			end
+			l_errors.force (utf.utf_32_string_to_utf_8_string_8 (a_code))
+		end
+
+	item (a_item: JSON_OBJECT): GCSE_PAGE_ITEM
+			-- Google Result Metadata Item.
+		do
+			create Result
+			if attached {JSON_STRING} a_item.item (create {JSON_STRING}.make_from_string ("kind")) as l_kind then
+				Result.set_kind (l_kind.item)
+			end
+			if attached {JSON_STRING} a_item.item (create {JSON_STRING}.make_from_string ("title")) as l_title then
+				Result.set_title (l_title.item)
+			end
+			if attached {JSON_STRING} a_item.item (create {JSON_STRING}.make_from_string ("htmlTitle")) as l_htmltitle then
+				Result.set_html_title (l_htmltitle.item)
+			end
+			if attached {JSON_STRING} a_item.item (create {JSON_STRING}.make_from_string ("link")) as l_link then
+				Result.set_link (l_link.item)
+			end
+			if attached {JSON_STRING} a_item.item (create {JSON_STRING}.make_from_string ("displayLink")) as l_display_link then
+				Result.set_display_link (l_display_link.item)
+			end
+			if attached {JSON_STRING} a_item.item (create {JSON_STRING}.make_from_string ("snippet")) as l_snippet then
+				Result.set_snippet (l_snippet.item)
+			end
+			if attached {JSON_STRING} a_item.item (create {JSON_STRING}.make_from_string ("htmlSnippet")) as l_html_snippet then
+				Result.set_html_snippet (l_html_snippet.item)
+			end
+			if attached {JSON_STRING} a_item.item (create {JSON_STRING}.make_from_string ("formattedUrl")) as l_formatted_url then
+				Result.set_formatted_url (l_formatted_url.item)
+			end
+		end
+
+	query_page (a_page_key: JSON_STRING; a_queries: JSON_OBJECT): detachable GCSE_PAGE
+			-- Google result medata query. Return a query page based for a query with page key `a_page_key', if any.
+		do
+			if attached {JSON_ARRAY} a_queries.item (a_page_key) as jquerypage and then attached {JSON_OBJECT} jquerypage.i_th (1) as jpage then
```

Comment:
you are right, at the moment we don't have a json_array.is_empty query, but we can have the same behavior using json_array.count > 0.